inject.js

```diff
@@ -436,15 +445,18 @@ function setupListener() {
 
     log("Updating controller with new speed", 5);
     speedIndicator.textContent = speed.toFixed(2);
-    tc.settings.speeds[src] = speed;
-    log("Storing lastSpeed in settings for the rememberSpeed feature", 5);
-    tc.settings.lastSpeed = speed;
-    log("Syncing chrome settings for lastSpeed", 5);
-    chrome.storage.sync.set({ lastSpeed: speed }, function () {
-      log("Speed setting saved: " + speed, 5);
-    });
-    // show the controller for 1000ms if it's hidden.
-    runAction("blink", null, null);
+    // when adaptiveSpeed is activated, there are too many changes.
```

Comment:
First off, love this! 

Could we make it work such that we don't have to choose between the two? As in, removing silence blocks seems like an additive option in addition to setting a base speed. Some podcasts apps provide this functionality, which you can toggle in addition to speed setting. 

I think we'd just have to distinguish ratechange events that are generated from this logic vs. other sources?

Reply:
I am totally with you. 
Just made an update. The problem is that ratechange is fired every time.
Now i made an attempt to distinguish before increasing or decreasing speed.

i think there is still a problem when turning off adaptive speed when in silent mode it seems to take the current "adapriveSpeed" as lastSpeed. so video stays at e.g. 5.00 .

Reply:
i also increased the timeout for the samples to 10ms to hopefully increase the preformance but i randumly seem to get async sound when using it alot.